File: vm/tests/all/interopComplex/inheritMember/inheritMember.js
//inheritMemberClass.js

class LinearFunction {
	
	constructor(kVal, dVal) {
		this.k=kVal;
		this.d=dVal;
	}

	evaluate(x) {
		return this.k*x+this.d;
	}
}

class ConstFunction extends LinearFunction {
	constructor(dVal) {
		super(0, dVal);
	}
}

File: vm/tests/all/interopComplex/inheritMember/inheritMember.cpp
//inheritMemberClass.cpp
#include<polyglot.h>
#include<stdio.h>


class LinearFunction {
	protected:
		int k;
		int d;
	public:
		LinearFunction(int k, int d);
		virtual int evaluate(int x);
};

POLYGLOT_DECLARE_CLASS(LinearFunction);

class ConstFunction: public LinearFunction {
	public:
		ConstFunction(int d);
};

POLYGLOT_DECLARE_CLASS(ConstFunction);

LinearFunction::LinearFunction(int kVal, int dVal) {
	k=kVal;
	d=dVal;
}

int LinearFunction::evaluate(x) {
	return k*x+d;
}

ConstFunction::ConstFunction(int dVal) : base(0, dVal) {}
	

